sections/shared/Layout/AppLayout/Header/BalanceActions.tsx

```diff
@@ -0,0 +1,257 @@
+import useSynthetixQueries from '@synthetixio/queries';
+import Button from 'components/Button';
+import CurrencyIcon from 'components/Currency/CurrencyIcon';
+import Select from 'components/Select';
+import { Synths } from 'constants/currency';
+import Connector from 'containers/Connector';
+import { useRouter } from 'next/router';
+import useGetFuturesMarkets from 'queries/futures/useGetFuturesMarkets';
+import useGetFuturesPositionForMarkets from 'queries/futures/useGetFuturesPositionForMarkets';
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import { FC } from 'react';
+import { useTranslation } from 'react-i18next';
+import { components } from 'react-select';
+import { useRecoilValue } from 'recoil';
+import { walletAddressState } from 'store/wallet';
+import styled, { useTheme } from 'styled-components';
+import { FlexDivRow, FlexDivRowCentered } from 'styles/common';
+import { formatCurrency, zeroBN } from 'utils/formatters/number';
+import { getDisplayAsset, getMarketKey } from 'utils/futures';
+
+type ReactSelectOptionProps = {
+	label: string;
+	synthIcon: string;
+	marketAccessibleMargin?: string;
+	onClick?: () => {};
+};
+
+type FuturesPositionTableProps = {
+	setShowUniswapWidget: Dispatch<SetStateAction<boolean>>;
+	uniswapWidgetOpened: boolean;
+};
+
+const BalanceActions: FC<FuturesPositionTableProps> = ({
+	uniswapWidgetOpened,
+	setShowUniswapWidget,
+}) => {
+	const [balanceLabel, setBalanceLabel] = useState('');
+	const { t } = useTranslation();
+	const theme = useTheme();
+	const router = useRouter();
+	const { network } = Connector.useContainer();
+
+	const walletAddress = useRecoilValue(walletAddressState);
+	const { useSynthsBalancesQuery } = useSynthetixQueries();
+	const synthsBalancesQuery = useSynthsBalancesQuery(walletAddress);
+	const sUSDBalance = synthsBalancesQuery?.data?.balancesMap?.[Synths.sUSD]?.balance ?? zeroBN;
+
+	const futuresMarketsQuery = useGetFuturesMarkets();
+	const futuresMarkets = futuresMarketsQuery?.data ?? [];
+	const futuresPositionQuery = useGetFuturesPositionForMarkets(
+		futuresMarkets.map(({ asset }) => getMarketKey(asset, network.id))
+	);
+	const futuresPositions = futuresPositionQuery?.data ?? [];
+
+	const accessiblePositions = futuresPositions.filter((position) =>
+		position.accessibleMargin.gt(zeroBN)
```

Comment:
I think it makes much more sense to use `remainingMargin` here, will describe why in my review comment.